### PR TITLE
Fix annotation trigger to reconcile on container image change

### PR DESCRIPTION
**Pull request by tnozicka**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1541685

@smarterclayton this should fix your issues with DS and annotation trigger

### CODE REVIEW

**Comment by tnozicka** [comment]
@openshift-ci-robot why would you request 2 more people when I've already requested review from 3 people?

@kargakis ? (you are the bot expert)

File: pkg/image/OWNERS [hunk 1]
```diff
@@ -11,3 +11,4 @@ approvers:
   - soltysh
   - mfojtik
   - liggitt
+  - tnozicka
```
Comment (kargakis) [inline pkg/image/OWNERS]: Add yourself as a reviewer too

**Comment by kargakis** [comment]
> @openshift-ci-robot why would you request 2 more people when I've already requested review from 3 people?

Unfortunately, the bot does not identify manual review requests today. If you used the bot's review request command (`/cc mfojtik ironcladlou smarterclayton`) in your starting post, it would have requested reviews only from the folks you wanted. Please open an issue in k8s.io/test-infra, we need to investigate whether it's possible to avoid automated review requests in case somebody has been manually requested to review.

File: pkg/image/OWNERS (hunk 1 above)
Comment (tnozicka) [reply pkg/image/OWNERS]: I though approver is a superset includung reviewer. I think I've already done it somewhere and lgtm works there. What's the difference?

**Comment by tnozicka** [comment]
@kargakis thanks for the explanation, I am fine using `/cc` from now on to avoid it. Don't want to add you more work.

File: pkg/image/OWNERS (hunk 1 above)
Comment (mfojtik) [reply pkg/image/OWNERS]: now you will have to do image api reviews \o/ :-)

File: pkg/image/trigger/annotations/annotations.go [hunk 2]
```diff
@@ -172,6 +172,33 @@ func UpdateObjectFromImages(obj runtime.Object, tagRetriever trigger.TagRetrieve
...
+// ContainerImageChanged returns true if any container image referenced by newTriggers changed.
+func ContainerImageChanged(oldObj, newObj runtime.Object, newTriggers []triggerapi.ObjectFieldTrigger) bool {
+	for _, trigger := range newTriggers {
+		if trigger.Paused {
+			continue
+		}
+
+		newContainer, _, err := ContainerForObjectFieldPath(newObj, trigger.FieldPath)
+		if err != nil {
+			glog.V(5).Infof("%v", err)
```
Comment (mfojtik) [inline pkg/image/trigger/annotations/annotations.go]: please add context for this error

File: pkg/image/trigger/annotations/annotations.go [hunk 3]
```diff
@@ -172,6 +172,33 @@ func UpdateObjectFromImages(obj runtime.Object, tagRetriever trigger.TagRetrieve
...
+			continue
+		}
+
+		oldContainer, _, err := ContainerForObjectFieldPath(oldObj, trigger.FieldPath)
+		if err != nil {
+			// might just be a result of the update
```
Comment (mfojtik) [inline pkg/image/trigger/annotations/annotations.go]: glog(5) here as well

File: pkg/image/trigger/annotations/annotations.go [hunk 4]
```diff
@@ -172,6 +172,33 @@ func UpdateObjectFromImages(obj runtime.Object, tagRetriever trigger.TagRetrieve
...
+			continue
+		}
+
+		if newContainer.GetImage() != oldContainer.GetImage() {
```
Comment (mfojtik) [inline pkg/image/trigger/annotations/annotations.go]: return newContainer.GetImage() != oldContainer.GetImage()

**Comment by mfojtik** [comment]
LGTM, will tag on green tests

File: pkg/image/OWNERS (hunk 1 above)
Comment (tnozicka) [reply pkg/image/OWNERS]: oh no, I am going to fall back

File: pkg/image/trigger/annotations/annotations.go (hunk 3 above)
Comment (tnozicka) [reply pkg/image/trigger/annotations/annotations.go]: if the triggers has changed or the images and the field path is valid only for the new image+trigger combination there will always be an error here although it's not an error in this case

**Pull request merged**